src/parser/TranspileState.ts

```diff
@@ -91,14 +70,46 @@ export class TranspileState {
      * because the entire token is passed by reference, instead of the raw string being copied to the parameter,
      * only to then be copied again for the SourceNode constructor
      */
-    public tokenToSourceNode(token: Token) {
+    private tokenToSourceNode(token: { range?: Range; text: string }) {
         return new SourceNode(
             //convert 0-based range line to 1-based SourceNode line
             token.range.start.line + 1,
             //range and SourceNode character are both 0-based, so no conversion necessary
             token.range.start.character,
-            this.pathAbsolute,
+            this.srcPath,
             token.text
         );
     }
+
+    /**
+     * Create a SourceNode from a token, accounting for missing range and multi-line text
+     */
+    public transpileToken(token: { range?: Range; text: string }) {
```

Comment:
This was mostly copied from SGTranspileState, except I converted the `map()` call to a for loop and call `this.SourceNode` to reuse existing sourceNode logic